llvm/include/llvm/ADT/SmallSet.h

```diff
@@ -234,19 +225,12 @@ class SmallSet {
   /// Check if the SmallSet contains the given element.
   bool contains(const T &V) const {
     if (isSmall())
-      return vfind(V) != Vector.end();
-    return Set.find(V) != Set.end();
+      return llvm::is_contained(Vector, V);
+    return llvm::is_contained(Set, V);
```

Comment:
Please do not use is_contained with fast set types.

Reply:
Thanks for pointing that out. I've checked and `llvm::is_contained` indeed does a linear scan on a `std::set` because the latter does not have a `contains` method before C++20.

Reply:
I'd expect it to call `set::find`: https://github.com/llvm/llvm-project/blob/49a754a43d5592e08ef177db794126ddc676d6b5/llvm/include/llvm/ADT/STLExtras.h#L1890-L1891

Reply:
My bad. @kuhar is right. It calls `std::set::find` which is as good as it gets with `std::set`.

Reply:
It doesn't matter what it actually calls, we should never use is_contained() for set structures, unless there is some very, very strong reason (e.g. generic context that really does not core whether it gets an O(1) set or O(n) set).

Reply:
This advice seems at least inconsistent with the intent of the code per the comment:
```
This is intended as the canonical way to check if an
element exists in a range in generic code or range type that does not
expose a `.contains(Element)` member.
```

In this case, the set meets the second requirement - that it does not expose a `contains` function?

Reply:
If you want to have an abstraction over contains() and find() != end(), I'd suggest adding another helper, which does not also include the linear scan case. Though I don't think this is really worthwhile, as the existing find() != end() code works fine.

Reply:
FWIW, I think is_contained is more legible than find!=end and seems fine to use here & I agree with the comment on is_contained suggesting it for uses like this.